### PR TITLE
Survey changes

### PR DESCRIPTION
**Pull request by JulReinhardt**
Gonna add all the changes we need to make to finalize the definition based on the survey results.

### CODE REVIEW

**Comment by prjemian** [comment]
@JulReinhardt Does this PR fix #53?  Is there something we can do on #53 in the next week?

**Comment by prjemian** [comment]
This branch needs to merge new content in the master branch.

**Comment by prjemian** [comment]
FWIW: merging changes from the master branch is a beast of a job this time

File: NeXus/NXxpcs.nxdl.xml [hunk 1]
```diff
@@ -38,6 +38,9 @@
   <doc>
     X-ray Photon Correlation Spectroscopy data (results).
 
+    It is expected that this XPCS data will be part of multi-modal data set that could involve e.f. NXcansas
+    or 1D and/or 2D data.
+
     see: https://docs.google.com/spreadsheets/d/1sZrhLlzER9nuEWmPSLbWcfsgH_0DlBI9QrGRBQdKRQg/edit#gid=0
```
Comment (prjemian) [inline NeXus/NXxpcs.nxdl.xml:44]: We can discard this link now/

File: NeXus/NXxpcs.nxdl.xml [hunk 2]
```diff
@@ -77,47 +80,92 @@
     </field>
 
     <group type="NXdata" name="data">
-      <field name="frame_sum" type="NX_NUMBER" units="NX_ANY">
+      <!-- TODO: -->
```
Comment (prjemian) [inline NeXus/NXxpcs.nxdl.xml:83]: This TODO refers to a need for some kind of documentation.

File: NeXus/NXxpcs.nxdl.xml [hunk 3]
```diff
@@ -77,47 +80,92 @@
     </field>
 
     <group type="NXdata" name="data">
-      <field name="frame_sum" type="NX_NUMBER" units="NX_ANY">
+      <!-- TODO: needs documentation -->
```
Comment (prjemian) [inline NeXus/NXxpcs.nxdl.xml:83]: This TODO is part of #41

File: NeXus/NXxpcs.nxdl.xml [hunk 4]
```diff
@@ -178,41 +228,7 @@
...
-              -->
-          <field name="mask" type="NX_NUMBER">
-            <doc>
-            ROI mask
-
-            The ``units`` attribute should be set to ``"au"``
-            indicating arbitrary units.
-            </doc>
-          </field>
+          <!--  FIXME: removed archaic doc text. -->
```
Comment (prjemian) [inline NeXus/NXxpcs.nxdl.xml:231]: Verify this FIXME comment can also be removed.

File: loader/nx_loader_aps.py [hunk 5]
```diff
@@ -23,18 +23,18 @@ def _get_c2t(self):
     def xpcs_md(self):
         xpcs = dict(
             g2=self.data_file.get("/exchange/norm-0-g2"),
-            g2_unit='a.u.',
+            g2_units='a.u.',
             g2_stderr=self.data_file.get("/exchange/norm-0-stderr"),
-            g2_stderr_unit='a.u.',
             tau=self.data_file.get("/exchange/tau"),
-            tau_unit='s',
+            tau_units='s',
```
Comment (prjemian) [inline loader/nx_loader_aps.py:29]: The NXDL does not call this tau now.  Is it to use `tau` here still?

File: creator/nx_writer_xpcs.py [hunk 6]
```diff
@@ -44,8 +44,8 @@ def go(self):
     # xpcs_md = {"g2": "g2_value",
     #            "g2_stderr": "g2_stderr_value",
     #            "tau": "tau_value",
-    #            "g2_partials_twotime": "g2_partials_twotime_values",
-    #            "g2_twotime": "g2_twotime_values",
+    #            "g2_from_two_time_corr_func_partials": "g2_partials_twotime_values",
+    #            "g2_from_two_time_corr_func": "g2_twotime_values",
     #            #TODO figure out how to access large amount of data
```
Comment (prjemian) [inline creator/nx_writer_xpcs.py:49]: Has this TODO item been resolved?

**Review by prjemian — CHANGES_REQUESTED** [review]
Some changes, some questions, as noted.

File: creator/nx_writer_xpcs.py (hunk 6 above)
Comment (JulReinhardt) [reply creator/nx_writer_xpcs.py:49]: I think at some point we discussed what array shape we would prefer but I can't recall how we wanted to deal with large numbers of g2 curves. I think we kind of left it open to whatever software would read the XPCS file. This TODO is probably for v2?

File: NeXus/NXxpcs.nxdl.xml (hunk 3 above)
Comment (JulReinhardt) [reply NeXus/NXxpcs.nxdl.xml:83]: done

**Comment by prjemian** [comment]
@JulReinhardt I believe this is ready for final review.  Can you check and confirm?

**Comment by JulReinhardt** [comment]
@prjemian Yes I think so. Since it's my PR, I would not merge myself.

**Comment by prjemian** [comment]
@JulReinhardt Most recently, I merged the newer changes from the master branch into this one.  Can you give it a final inspection to make sure it did not introduce a problem from that merge (it was complicated).

**Comment by prjemian** [comment]
@JulReinhardt The merge from the master branch (to resolve conflicts) was easier than I expected.  It is now complete.  Please check it over to see if I have dropped anything while resolving conflicts.

**Review by prjemian — APPROVED** [review]

**Comment by prjemian** [comment]
@JulReinhardt - I recycled my approval for one last review after you check if anything is missing.

**Comment by JulReinhardt** [comment]
@prjemian looking through it right now, will approve any time soon.

**Comment by JulReinhardt** [comment]
@prjemian looks all good to me, we have some TODOs and FIXMEs left as comments in the xml file. Other than that it's ready to merge.

**Review by prjemian — APPROVED** [review]
LGTM!  @AbbyGi If you approve, feel free to merge.

@JulReinhardt Big thanks!

**Review by AbbyGi — APPROVED** [review]
Looks good. Merging

**Comment by prjemian** [comment]
@AbbyGi Thanks!

**Pull request merged**